src/gamecontrol.h

```diff
@@ -10,13 +10,102 @@
 #ifndef GAMECONTROL_H
 #define GAMECONTROL_H
 
-void StartNewGame();
-void ShutdownGame();
-
 void OnNewDay();
 void OnNewMonth();
 void OnNewYear();
 void OnNewFrame(uint32 frame_delay);
 
-#endif
+/** Actions that can be run to control the game. */
+enum GameControlAction {
+	GCA_NONE,      ///< No action to run.
+	GCA_NEW_GAME,  ///< Prepare a new game.
+	GCA_LOAD_GAME, ///< Load a saved game.
+	GCA_SAVE_GAME, ///< Save the current game.
+	GCA_QUIT,      ///< Quit the game.
+};
+
+/**
+ * Class controlling the current game.
+ */
+class GameControl {
+public:
+	GameControl();
+	~GameControl();
+
+	void RunAction();
+	void NewGame();
+	void LoadGame(const char *fname);
+	void SaveGame(const char *fname);
+	void QuitGame();
+
+	bool running;
```

Comment:
It may be worth having an Initialize method that sets this boolean, as well as handles starting a new game, saved game, etc. This would replace the _game_control.NewGame() call that is made prior to _video.MainLoop(). With the added Initialize(), I can also add an Uninitialize that would come after the MainLoop, which would most likely run ShutdownLevel for now.